dataset/src/test/scala/frameless/ops/SmartProjectTest.scala

```diff
@@ -12,7 +12,7 @@ case class InvalidFooProjectionType(i: Int, x: Boolean)
 case class InvalidFooProjectionName(i: Int, xerr: String)
 
 class SmartProjectTest extends TypedDatasetSuite {
-  val dataset = TypedDataset.create(Foo(1, 2, "hi") :: Foo(2, 3, "there") :: Nil)
+  lazy val dataset = TypedDataset.create(Foo(1, 2, "hi") :: Foo(2, 3, "there") :: Nil)
```

Comment:
Could you add a comment in the source to explain why laziness is required here?

Reply:
@Voltir thanks for the edit! I like the organization for the tests. However, the lazy here looks kind of strange.

Reply:
```
// Lazy needed to prevent initialization anterior to the `beforeAll` hook
```

Reply:
I don't get it :/ ... why just that one line then? we have so many initializations in our tests and I don't see anything special with that particular line ... what am I missing here?

Reply:
What Olivier said - i made SparkTest less safe by using a private var initialized to null, which is then properly initialized in the beforeAll test hook. This particular val is initialized before "beforeAll" and this will blow up with a null pointer exception - marking it lazy defers evaluation until its first use - inside a test that runs after "beforeAll", making everything work out.

I think this is the only place a dataset is initialized outside of a test, so this is the only place that had to be marked lazy.